src/backend/d3d12/RenderPassDescriptorD3D12.cpp

```diff
@@ -0,0 +1,78 @@
+// Copyright 2017 The NXT Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "backend/d3d12/RenderPassDescriptorD3D12.h"
+
+#include "backend/d3d12/D3D12Backend.h"
+#include "backend/d3d12/TextureD3D12.h"
+#include "common/BitSetIterator.h"
+
+namespace backend { namespace d3d12 {
+
+    RenderPassDescriptor::RenderPassDescriptor(Device* device, RenderPassDescriptorBuilder* builder)
+        : RenderPassDescriptorBase(builder), mDevice(device) {
+        // Get and fill an RTV heap with the color attachments
+        uint32_t colorAttachmentCount = static_cast<uint32_t>(GetColorAttachmentMask().count());
+        if (colorAttachmentCount != 0) {
+            mRtvHeap = device->GetDescriptorHeapAllocator()->AllocateCPUHeap(
+                D3D12_DESCRIPTOR_HEAP_TYPE_RTV, colorAttachmentCount);
+        }
+
+        for (uint32_t i : IterateBitSet(GetColorAttachmentMask())) {
```

Comment:
I think it would be clearer to move this `for` loop inside the preceding `if` block